### PR TITLE
coordinator: use random key for intermediate CA

### PR DESCRIPTION
**Pull request by burgerdev**
When we introduced Coordinator recovery, we decided to derive all key material deterministically from the seed. This implies that whoever owns the seed can derive these keys offline, too. However, since the seed needs to be held by the workload owner for recovery, the data owner can't trust in the mesh CA cert without trusting the workload owner.

This PR reverts to a random mesh/intermediate key, which is inaccessible to the workload owner. The trade-off is that the mesh key changes after recovery, and all workloads need to be restarted in order to pick up this change.

### CODE REVIEW

File: e2e/openssl/openssl_test.go [hunk 1]
```diff
@@ -201,15 +201,24 @@ func TestOpenSSL(t *testing.T) {
...
+			if err != nil {
+				t.Logf("openssl with %q after recovery:\n%s", rootCAFile, stdout)
+			}
+			assert.NoError(t, err, "stderr: %q", stderr)
+		})
+
+		require.NoError(c.Restart(ctx, kubeclient.Deployment{}, ct.Namespace, opensslBackend))
+		require.NoError(c.WaitFor(ctx, kubeclient.Deployment{}, ct.Namespace, opensslBackend))
+
+		t.Run("mesh CA after coordinator recovery is accepted when workloads are restarted", func(t *testing.T) {
```
Comment (katexochen) [inline e2e/openssl/openssl_test.go:215]: Should we also check mesh CA is not accepted before the restart?

**Review by katexochen — APPROVED** [review]
lgtm, thanks!

File: e2e/openssl/openssl_test.go (hunk 1 above)
Comment (burgerdev) [reply e2e/openssl/openssl_test.go:220]: Makes sense, added a test.

**Comment by edgelessci** [comment]
Successfully created backport PR for `release/v0.8`:
- #733

**Pull request merged**